src/command/verify/verify.c

```diff
@@ -1840,21 +2172,52 @@ verifyProcess(const bool verboseText)
 
                 // ??? Need to do the final reconciliation - checking backup required WAL against, valid WAL
 
-                // Report results
-                resultStr = verifyRender(jobData.archiveIdResultList, jobData.backupResultList, verboseText);
+                // Prepare results KV
+                resultKv = verifyPrepareResult(jobData.archiveIdResultList, jobData.backupResultList, verboseText);
             }
             else if (!backupLabelInvalid)
+            {
                 strCatZ(resultStr, "\n    no archives or backups exist in the repo");
+                verifyMessageNew(errorList, printNever,
+                                 strNewZ("no archives or backups exist in the repo"));
+            }
 
             errorTotal += jobData.jobErrorTotal;
         }
 
-        // If verbose output or errors then output results
-        if (verboseText || errorTotal > 0)
+        if (resultKv == NULL)
+            resultKv = kvNew();
```

Comment:
Maybe just do `resultKv = kvNew();`  and send it into verifyPrepareResult() to avoid this checking and creation resultKv into verifyPrepareResult?
It would be safer because it would not create external resultKv that not to forget to clean.